graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_hf_transformation.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import Sequence
+from pyspark.sql import DataFrame
+from pyspark.sql.types import ArrayType, IntegerType, StructType, StructField
+from pyspark.sql.functions import udf
+
+import numpy as np
+from transformers import AutoTokenizer
+
+from graphstorm_processing.constants import HUGGINGFACE_TOKENIZE
+from .base_dist_transformation import DistributedTransformation
+
+
+def apply_norm(
+    cols: Sequence[str], bert_norm: str, bert_model: str, max_seq_length: int, input_df: DataFrame
+) -> DataFrame:
+    """Applies a single normalizer to the imputed dataframe, individually to each of the columns
+    provided in the cols argument.
+
+    Parameters
+    ----------
+    cols : Sequence[str]
+        List of column names to apply normalization to.
+    bert_norm : str
+        The type of normalization to use. Valid values is "tokenize"
+    bert_model : str
+        The name of huggingface model.
+    input_df : DataFrame
+        The input DataFrame to apply normalization to.
+    """
+
+    if bert_norm == HUGGINGFACE_TOKENIZE:
+        # Initialize the tokenizer
+        tokenizer = AutoTokenizer.from_pretrained(bert_model)
+
+        # Define the schema of your return type
+        schema = StructType(
+            [
+                StructField("input_ids", ArrayType(IntegerType())),
+                StructField("attention_mask", ArrayType(IntegerType())),
+                StructField("token_type_ids", ArrayType(IntegerType())),
+            ]
+        )
+
+        # Define UDF
+        @udf(returnType=schema)
```

Comment:
Can we determine Spark version somehow and use arrow UDFs when available?

Reply:
With the low Spark version, defining a UDF with @udf(returnType=schema, useArrow=True) is even illegal and throws error during compiler stage. So I would suggest to move to arrow UDFs once Spark 3.5 is all available.

Reply:
Yes, I was talking about something like:

```python
import pyspark
from packaging.version import parse as version_parse

if version_parse(pyspark.__version__) >= version_parse("3.5.0")
    # Use arrow UDF
else:
    # Use plain UDF
```

Might be too complex though, so perhaps we'll leave arrow support for the next release.

Reply:
Agree. I would suggest to do it later, and we can apply it to all kinds of feature transformation.